recipes/ffmpeg/all/conanfile.py

```diff
@@ -0,0 +1,348 @@
+                       "lzma": True,
+                       "iconv": True,
+                       "freetype": True,
+                       "openjpeg": True,
+                       "openh264": True,
+                       "opus": True,
+                       "vorbis": True,
+                       "zmq": False,
+                       "sdl2": False,
+                       "x264": True,
+                       "x265": True,
+                       "vpx": True,
+                       "mp3lame": True,
+                       "fdk_aac": True,
+                       "webp": True,
+                       "openssl": True,
+                       "alsa": True,
+                       "pulse": True,
+                       "vaapi": True,
+                       "vdpau": True,
+                       "xcb": True,
+                       "appkit": True,
+                       "avfoundation": True,
+                       "coreimage": True,
+                       "audiotoolbox": True,
+                       "videotoolbox": True,
+                       "securetransport": False,  # conflicts with OpenSSL
+                       "qsv": True}
+    generators = "pkg_config"
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc" and os.name == "nt"
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+            destination=self._source_subfolder, strip_root=True)
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os != "Linux":
+            del self.options.vaapi
+            del self.options.vdpau
+            del self.options.xcb
+            del self.options.alsa
+            del self.options.pulse
+        if self.settings.os != "Macos":
+            del self.options.appkit
+            del self.options.avfoundation
+            del self.options.coreimage
+            del self.options.audiotoolbox
+            del self.options.videotoolbox
+            del self.options.securetransport
+        if self.settings.os != "Windows":
+            del self.options.qsv
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def build_requirements(self):
+        self.build_requires("yasm/1.3.0")
+        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
+        if self.settings.os == "Linux":
+            if not tools.which("pkg-config"):
+                self.build_requires("pkgconf/1.7.4")
```

Comment:
Why Linux only? Are you sure that upstream build system doesn't try to find dependencies with pkg_config?
Not sure that tools.which is recommended, maybe we want to have full control of pkgconf.